hack/update-openapigen.sh

```diff
@@ -22,17 +22,38 @@ if [ -z "${GOPATH:-}" ]; then
     export GOPATH=$(go env GOPATH)
 fi
 
-# Update kative files to add `+k8s:openapi-gen=true` to workaround the knative types cannot be load.
-# TBD @jinchihe: This should be updated from knative (PR: https://github.com/knative/pkg/pull/510).
-sed -i '/^type Condition struct {/i // +k8s:openapi-gen=true' vendor/github.com/knative/pkg/apis/condition_types.go
-sed -i '/^type VolatileTime struct {/i // +k8s:openapi-gen=true' vendor/github.com/knative/pkg/apis/volatile_time.go
+if [ -z "${GOROOT:-}" ]; then
+    export GOROOT=$(go env GOROOT)
+fi
+
+# Below hacks are to add "// +k8s:openapi-gen=true" for some dependencies.
+# The knative related hacking can be removed once upgraded (knative/pkg/pull/510).
+knative_condition_file="vendor/github.com/knative/pkg/apis/condition_types.go"
+knative_volatile_file="vendor/github.com/knative/pkg/apis/volatile_time.go"
+knative_url_file="vendor/github.com/knative/pkg/apis/url.go"
```

Comment:
should we create our own url type ? Looks it generates a bunch of fields that we do not need.

Reply:
Hi @yuzisun That's needed, Yes, we can use that `URL` struct directly, but in the URL struct, there is a [Userinfo](https://github.com/golang/go/blob/a2f5d644d3a2903a82e527e3bca2c4bdff1d0750/src/net/url/url.go#L359) that's linked to another struch in the file. If not creating the url type, there will be have import traceback [here](https://github.com/kubeflow/kfserving/blob/8ddc3f027187134f241888c0dda1572ff06533ac/sdk/kfserving/models/knative_url.py#L19) and no link for the document [KnativeURL.md](https://github.com/kubeflow/kfserving/blob/8ddc3f027187134f241888c0dda1572ff06533ac/sdk/docs/KnativeURL.md)